### PR TITLE
[App Service] BREAKING CHANGE: `az webapp config`: Redact webapp config connection-string and storage-account values

### PR DESCRIPTION
**Pull request by amberwang113**
**Related command**
az webapp config connection-string set/delete
az webapp config storage-account add/update/delete

**Description**
Necessary to protect potential passwords in webapp connection strings and storage account access keys. Redacts outputs of set/delete (connection-string) by setting value for each added connection string to null. Redacts outputs of add/update/delete (storage-account) by setting accesskey for each added storage account to null. Added warning to indicate that behavior has changed and that users should use az webapp config connection-string/storage-account list to view values of the connection strings.

**Testing Guide**
For connection-string:
1. Create webapp (az create webapp)
2. Add a connection string (az webapp config connection-string set)

azdev test test-webapp-config
azdev test test-webapp-slot

For storage-account:
1. Create Linux webapp (az webapp create with Linux plan)
2. Link webapp to storage account (az webapp config storage-account add)

azdev test test_webapp_config

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -1686,9 +1686,18 @@ def delete_connection_strings(cmd, resource_group_name, name, setting_names, slo
...
-                                       conn_strings, slot, client)
+    result = _generic_settings_operation(cmd.cli_ctx, resource_group_name, name,
+                                         'update_connection_strings',
+                                         conn_strings, slot, client)
+    _redact_connection_strings(result.properties)
+    return result
+
+
+def _redact_connection_strings(properties):
+    logger.warning('Connection string values have been redacted. Use config connection-string list to view.')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:1697]: Same as https://github.com/Azure/azure-cli/pull/27565/files#r1364909762

**Comment by yanzhudd** [comment]
Please rerun the test and upload the recording file.

**Comment by yanzhudd** [comment]
It is suggested to specify the related command in the title:
[App Service] BREAKING CHANGE: `az webapp config`: Redact webapp config connection-string and storage-account values

**Comment by amberwang113** [comment]
Re-recorded the two affected tests and fixed title!

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by yanzhudd** [comment]
Please resolve the code conflicts

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**